### PR TITLE
Cached data

### PR DESCRIPTION
**Pull request by kdobolyi**
Asking to merge the changes that include caching of the data and bug fixes to the main branch.

### CODE REVIEW

**Comment by jspeed-meyers** [comment]
@kdobolyi, I apologize for my slow review. Thank you for working on all these features. First question before I review this though, why is there a conflict? Would you be willing to resolve the conflict first so that I can review after that?

File: aura-borealis-flask-app/app.py [hunk 1]
```diff
@@ -1,35 +1,26 @@
 from flask import Flask, render_template, request, make_response, jsonify
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/app.py:1]: Okay, this proposed change would help me in the future, though I admit it is nit-picky. Can you please follow the proposed import order for packages from the Python style guide, aka PEP 8? A link is here: https://www.python.org/dev/peps/pep-0008/#imports

File: aura-borealis-flask-app/app.py [hunk 2]
```diff
@@ -1,35 +1,26 @@
...
 import json
 import time
 import pickle
-import os.path
-import urllib.parse as urlparse
-from urllib.parse import parse_qs
 
 from search import PackageSearch
 import os
 SECRET_KEY = os.urandom(32)
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/app.py:10]: Can you please add a comment that explains why the app needs a secret key?

File: aura-borealis-flask-app/app.py [hunk 3]
```diff
@@ -122,20 +109,25 @@ def about():
 # Display all packages with the highest number of warnings (uses live data)
 @app.route('/top_warnings/', methods=['GET', 'POST'])
 def top_warnings():
-	# load the live Aura data
+
 	warning_types_selected = []
 
 	if request.method == 'POST':
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/app.py:115]: Can you please add a comment that explains this if-else logic?

File: aura-borealis-flask-app/app.py [hunk 4]
```diff
@@ -376,59 +359,76 @@ def comparison():
...
-                    cleaned_locs.append({'warning_type':warning_type, 'severity':severity, 'line':loc[1], 'code':loc[0], 'filename':loc[2]})
-                 else:   
-                    cleaned_locs.append({'warning_type':warning_type, 'severity':severity, 'line':loc[1], 'code':loc[0], 'filename':loc[2].split('$')[1]})
-         if found_non_empty:
-            data.append(entry)
-
-     loc_columns = [
-	 {
+
+	warning_types_selected = []
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/app.py:363]: Can you please add a few comments in here so my future self can understand the code?

File: aura-borealis-flask-app/app.py [hunk 5]
```diff
@@ -376,59 +359,76 @@ def comparison():
...
+		for severity in SEVERITIES:
+			cleaned_locs = []
+			for loc in LOCs:
+				if loc[2] is None:
+					cleaned_locs.append({'warning_type':warning_type, 'severity':severity, 'line':loc[1], 'code':loc[0], 'filename':loc[2]})
+				else:   
+					cleaned_locs.append({'warning_type':warning_type, 'severity':severity, 'line':loc[1], 'code':loc[0], 'filename':loc[2].split('$')[1]})
+
+	entries = {}
+	for loc in LOCs:
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/app.py:396]: Same here please. A few comments.

File: aura-borealis-flask-app/app.py [hunk 6]
```diff
@@ -376,59 +359,76 @@ def comparison():
...
+			for severity in SEVERITIES:
+				if severity in entries[warning_type].keys(): 
+					row[severity] = entries[warning_type][severity]['count']
+				else:
+					row[severity] = 0
+
+			data.append(row)
+
+	cleaned_locs = []
+	LOCs = get_LOC_by_warning(package)
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/app.py:422]: Same here please.

File: aura-borealis-flask-app/app.py [hunk 7]
```diff
@@ -488,66 +488,68 @@ def single_package():
...
-    #unique_packages = get_unique_package_list()
-    #for pkg in unique_packages:
-    #    pkg_list.extend(map(lambda st: st.strip(), map(lambda s: str(s), pkg.split(','))))
-    filtered=filter(lambda ing: ing.startswith(inp),set(unique_packages))
-    #print(list(filtered))
-    return jsonify({"listaing":list(filtered)})
-    #return make_response({"listaing":list(filtered)}, 200)
+	pkg_list = []
+	filtered=filter(lambda ing: ing.startswith(inp),set(unique_packages))
+	return jsonify({"listaing":list(filtered)})
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/app.py:505]: What is "listaing"? That is a strange string to me.

File: aura-borealis-flask-app/live_data.py [hunk 8]
```diff
@@ -329,62 +322,59 @@ def get_all_warnings_counts_x(warning_type, all_warnings, all_unique_warnings, a
...
-	get all the warnings for a warning_type
-
-	Arguments
-	---------
-	warning_type : the Aura warning type you care about
-	all_warnings : a dictionary of all warnings, grouped by package, to be updated
-	all_unique_warnings : a dictionary of all unique warnings, grouped by package, to be updated
-	all_severities :  dictionary of severity scores, grouped by package, to be updated
-	"""
+    """
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/live_data.py:325]: Thank you for these helpful doctstrings.

File: aura-borealis-flask-app/live_data2.py [hunk 9]
```diff
@@ -0,0 +1,397 @@
+"""
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/live_data2.py:1]: What is this live_data2.py file? How is it different from live_data.py? Can you please put in a comment at the top of both files that explains the difference.

File: aura-borealis-flask-app/live_data2.py [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+		print("in ruc")
+		print(row[0])
+		results.append([row[0], row[1], row[2]])
+	return results
+
+
+# http://192.168.68.9:5601/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-1y,to:now))&_a=(columns:!(_source),filters:!(),index:'30bc7830-7530-11eb-870c-ad2473be90c7',interval:auto,query:(language:kuery,query:''),sort:!())
+
+
+def get_all_warnings_x(warning_type, all_warnings):
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/live_data2.py:120]: Can you please add in some function docstrings and comments?

File: aura-borealis-flask-app/templates/top_warnings.html [hunk 11]
```diff
@@ -41,9 +42,39 @@
       });
 
     });
+
+    function fillData(){
```
Comment (jspeed-meyers) [inline aura-borealis-flask-app/templates/top_warnings.html:46]: Can you add one comment to explain this section to a donut, ie me?

**Review by jspeed-meyers — CHANGES_REQUESTED** [review]
Hi @kdobolyi, great work. I left some comments, but they are all almost the same. Could you please add more comments, both function docstrings and in-line comments? This will help me when you, myself, or some future engineer looks at this code. No need to go crazy, but I do think there are some parts that would benefit. Feel free to push back if you think I am asking for excessive documentation.

File: aura-borealis-flask-app/app.py (hunk 7 above)
Comment (kdobolyi) [reply aura-borealis-flask-app/app.py:505]: me too -- we should ask @monag-ucberkeley

File: aura-borealis-flask-app/live_data2.py (hunk 9 above)
Comment (kdobolyi) [reply aura-borealis-flask-app/live_data2.py:1]: it was a backup, deleting, thanks!

**Pull request merged**